pkg/doc/did/doc.go

```diff
@@ -166,9 +166,35 @@ type Service struct {
 	Properties      map[string]interface{}
 }
 
+// VerificationRelationship defines a verification relationship between DID subject and a verification method.
+type VerificationRelationship int
+
+const (
+	// VerificationRelationshipGeneral is a special case of verification relationship: when a public key
+	// defined in PublicKey is not used by any VerificationMethod.
+	VerificationRelationshipGeneral VerificationRelationship = iota
+
+	// Authentication defines verification relationship.
+	Authentication
+
+	// AssertionMethod defines verification relationship.
+	AssertionMethod
+
+	// CapabilityDelegation defines verification relationship.
+	CapabilityDelegation
+
+	// CapabilityInvocation defines verification relationship.
+	CapabilityInvocation
+
+	// KeyAgreement defines verification relationship.
+	KeyAgreement
+)
+
 // VerificationMethod authentication verification method
 type VerificationMethod struct {
-	PublicKey PublicKey
+	PublicKey    PublicKey
```

Comment:
I assume that you accounted for the fact that a PublicKey can appear (by reference) in multiple sections.

Reply:
Yes, it's accounted by `Embedded` flag.